### PR TITLE
Change authentication stages

### PR DESCRIPTION
**Pull request by gumityolcu**
Endpoints in files _auth.js_, _index.js_, _search.js_ and _upload.js_ don't require JWT authentication.

POST endpoint for adding media to events doesn't require authentication because i think it would be considerably reasonable to do that.

GET endpoint for getting attendance info requires authentication because it would be reasonably inconsiderate not to do that.

Other than these exceptions, all PUT, DELETE and POST endpoints require authentication whereas no GET endpoint does.

### CODE REVIEW

**Comment by okyksl** [comment]
@nazmicancalik

File: backend/routes/events.js [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
+router.put('/:id/media/:mediaId', passport.authenticate('jwt', {session: false}), EventController.updateMedia);
+router.delete('/:id/media/:mediaId', passport.authenticate('jwt', {session: false}), EventController.deleteMedia);
 
 
 // Adds an attendance to an event
-router.post('/:id/attendance', EventController.addAttendance);
-router.get('/:id/attendance', EventController.getAttendance);
-router.put('/:id/attendance', EventController.updateAttendance);
+router.post('/:id/attendance', passport.authenticate('jwt', {session: false}), EventController.addAttendance);
+router.get('/:id/attendance', passport.authenticate('jwt', {session: false}), EventController.getAttendance);
```
Comment (okyksl) [inline backend/routes/events.js]: Is this **get** still needs to be private to users?

File: backend/routes/events.js [hunk 2]
```diff
@@ -1,40 +1,41 @@
...
 const EventController = require('../controllers/EventController');
 
 router.get('/', EventController.getAllEvents);
 router.get('/getEventbyCreator/:id', EventController.getEventbyCreator);
 
-router.post('/', EventController.addEvent);
+router.post('/', passport.authenticate('jwt', {session: false}), EventController.addEvent);
 
 //Media endpoints
 router.post('/:id/media', EventController.addMedia);
```
Comment (okyksl) [inline backend/routes/events.js]: I guess you missed **addMedia**.

File: backend/routes/events.js (hunk 1 above)
Comment (gumityolcu) [reply backend/routes/events.js]: We shouldn't share information about where users will be to non members. I did this intuitively.

Comment (gumityolcu) [reply backend/routes/events.js]: We shouldn't share information about where users will be to non members. I did this intuitively.

File: backend/routes/events.js (hunk 2 above)
Comment (gumityolcu) [reply backend/routes/events.js]: No, I explaines this in the PR message 😁

Comment (okyksl) [reply backend/routes/events.js]: Sorry for the unattended review :((

File: backend/routes/events.js (hunk 1 above)
Comment (okyksl) [reply backend/routes/events.js]: Is it still possible for us to retrieve **attendance counts**?

**Comment by okyksl** [comment]
We will integrate changes first to the `staging`, then to `master`.

**Pull request closed without merging**